src/Native/Unix/System.IO.Ports.Native/pal_termios.c

```diff
@@ -0,0 +1,356 @@
+    switch (speed)
+    {
+    case 230400:
+        return B230400;
+    case 115200:
+        return B115200;
+    case 57600:
+        return B57600;
+    case 38400:
+        return B38400;
+    case 19200:
+        return B19200;
+    case 9600:
+        return B9600;
+    case 4800:
+        return B4800;
+    case 2400:
+        return B2400;
+    case 1800:
+        return B1800;
+    case 1200:
+        return B1200;
+    case 600:
+        return B600;
+    case 300:
+        return B300;
+    case 200:
+        return B200;
+    case 150:
+        return B150;
+    case 134:
+        return B134;
+    case 110:
+        return B110;
+    case 75:
+        return B75;
+    case 50:
+        return B50;
+    }
+    return ((speed_t)(speed));
+}
+
+static int
+TermiosRate2Speed(speed_t brate)
+{
+    switch (brate)
+    {
+    case B230400:
+        return 230400;
+    case 115200:
+        return 115200;
+    case B57600:
+        return 57600;
+    case B38400:
+        return 38400;
+    case B19200:
+        return 19200;
+    case B9600:
+        return 9600;
+    case B4800:
+        return 4800;
+    case B2400:
+        return 2400;
+    case B1800:
+        return 1800;
+    case B1200:
+        return 1200;
+    case B600:
+        return 600;
+    case B300:
+        return 300;
+    case B200:
+        return 200;
+    case B150:
+        return 150;
+    case B134:
+        return 134;
+    case B110:
+        return 110;
+    case B75:
+        return 75;
+    case B50:
+        return 50;
+    }
+
+    return brate;
+}
+
+int32_t
+TermiosGetSpeed(int32_t fd)
+{
+    struct termios       term;
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    return TermiosRate2Speed(cfgetispeed(&term));
+}
+
+int32_t
+TermiosSetSpeed(int32_t fd, uint32_t speed)
+{
+    struct termios       term;
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    cfsetspeed(&term, TermiosSpeed2Rate(speed));
+
+    if (tcsetattr(fd, TCSANOW, &term) < 0)
+    {
+        return -2;
+    }
+
+    return speed;
+}
+
+int32_t
+TermiosAvailableBytes(int32_t fd, int32_t readBuffer)
+{
+    int32_t bytes;
+    if (ioctl (fd, readBuffer ? FIONREAD : TIOCOUTQ, &bytes) == -1) {
+        return -1;
+    }
+
+    return bytes;
+}
+
+int32_t
+TermiosDiscard(int32_t fd, int32_t queue)
+{
+    return tcflush(fd, queue == 0 ? TCIOFLUSH : queue == 1 ? TCIFLUSH : TCOFLUSH);
+}
+
+int32_t
+TermiosDrain(int32_t fd)
+{
+    return tcdrain(fd);
+}
+
+int32_t
+TermiosSendBreak(int32_t fd, uint32_t duration)
+{
+    return tcsendbreak(fd, duration);
+}
+
+int32_t
+TermiosReset(int32_t fd, int speed, int dataBits, int stopBits, int parity, int handshake)
+{
+    struct termios       term;
+    int ret = 0;
+
+    if (tcgetattr(fd, &term) < 0)
+    {
+        return  -1;
+    }
+
+    cfmakeraw(&term);
+    term.c_cflag |=  (CLOCAL | CREAD);
+    term.c_lflag &= ~((tcflag_t)(ICANON | ECHO | ECHOE | ECHOK | ECHONL | ISIG | IEXTEN ));
```

Comment:
I wonder - why don't we set all the flags ourselves and depend on the preinitialization by the cfmakeraw instead?